### ISSUE TITLE
OpenUI5 detection does not work

### ISSUE THREAD

**Issue by maxibanki** [AliasIO/Wappalyzer#1906]
Hi, 

example page: https://sapui5.hana.ondemand.com/demoapps.html

on which the `sap.ui` javascript variable is set. (check dev tools). See:

https://github.com/AliasIO/Wappalyzer/blob/123058f69b5738dcb156ed145e4c69d7d51cd4d9/src/apps.json#L6189-L6196

But strangely it is not working.

**Comment by gadcam** [comment]
I am sorry I did not understand that you were trying to match a property of an object. It is a normal behavior in fact see [the docs](https://www.wappalyzer.com/docs/specification). 

> Global JavaScript variables.
Members of the window object only; "^jQuery.fn$" won't work. Avoid short patterns to prevent matching minified code.

However maybe you could replace it with <code>sap-ui-(?!config|debug|optimized)</code> ?

**Comment by maxibanki** [comment]
Fact is, that the detection does not work. Maybe it can't check nested objects? Can I only check if the global JavaScript variable `sap` is available?

**Comment by gadcam** [comment]
I think you did not have the time to read my comment :)
By the way if you change the patten maybe we could add version detection with this pattern, what do you think ?
```json
"script": "^https://openui5\\.hana\\.ondemand\\.com/([\\d.]+)?/?resources/sap-ui-core.js$\\;version:\\1""
```

**Comment by gadcam** [comment]
Also I just checked something : you didn't add a script pattern for all the websites so I think can also make the part before "resources/" (and including it) optional.

**Comment by maxibanki** [comment]
Yes, sorry. I was just looking to less over your comment.
Great idea to check for the `sap-ui-core.js`!
We develop Open UI5 at work and in my experience there is always an `sap-ui-core.js` there, so we can check for that.

**Comment by maxibanki** [comment]
Fixed in 18733b6c5f755cbd6fe36f8c051e8bee16af0ecd
